PickMeUp/PickMeUpTests/PMNodeTests.m

```diff
@@ -0,0 +1,171 @@
+//
+//  PMNodeTests.m
+//  PickMeUpTests
+//
+//  Created by Abigail Shilts on 7/21/22.
+//
+
+#import <XCTest/XCTest.h>
+#import "PMNode.h"
+
+@interface PMNodeTests : XCTestCase
+@property (nonatomic, strong) PMNode *testingEmptChildren;
+@property (nonatomic, strong) PMNode *testingOneChild;
+@property (nonatomic, strong) PMNode *testingTwoChildren;
+
+@end
+
+@implementation PMNodeTests
+
+- (void)setUp {
+    self.testingEmptChildren = [PMNode new];
+    self.testingOneChild = [PMNode new];
+    self.testingTwoChildren = [PMNode new];
+    
+    self.testingEmptChildren.prefix = @"abc";
+    self.testingOneChild.prefix = @"abd";
+    self.testingTwoChildren.prefix = @"abe";
+}
+
+// Test focusing on adding a node to an empty child array
+- (void)testWhenEmpty {
+    PMNode *toAdd = [PMNode new];
+    toAdd.prefix = @"abcd";
+    [self.testingEmptChildren setChild:toAdd];
+    NSMutableArray<PMNode *> *testingAgainst = [NSMutableArray array];
+    [testingAgainst addObject:toAdd];
+    NSArray<PMNode *> *testingFor = [self.testingEmptChildren getChildren];
+    XCTAssertEqual(testingFor.count, testingAgainst.count);
+    int i;
+    for (i = 0; i < testingFor.count; i++) {
```

Comment:
no need to declare i on the separate line
``for (int i = 0...


Also in objective c you can iterate over array using this syntax
```
for (PMNode node in array) {
    // do something with object
}
```